main/server.go

```diff
@@ -0,0 +1,305 @@
+// Copyright 2014 Google Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to writing, software distributed
+// under the License is distributed on a "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build appenginevm
+// This package implements a simple HTTP server providing a REST API to a todo
+// handler.
+//
+// It provides six methods:
+//
+// 	GET	/todos		Retrieves all the todos.
+// 	POST	/todos		Creates a new todo given a title.
+//	DELETE	/todos		Deletes all completed todos.
+// 	GET	/todos/{todoID}	Retrieves the todo with the given id.
+// 	PUT	/todos/{todoID}	Updates the todo with the given id.
+//	DELETE	/todos/{todoID}	Deletes the todo with the given id.
+//
+// Every method below gives more information about every API call, its
+// parameters, and its results.
+package main
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+
+	"github.com/GoogleCloudPlatform/gcloud-golang-todos/todo"
+	"github.com/gorilla/mux"
+	"golang.org/x/net/context"
+	"google.golang.org/appengine"
+	"google.golang.org/appengine/log"
+)
+
+const PathPrefix = "/api/todos"
+const SlashedPathPrefix = PathPrefix + "/"
```

Comment:
everywhere you use this constant you also append a string, which might as well begin with "/". the constant isn't pulling its weight here.